### PR TITLE
Fixes the hybrid retriever query

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

This PR fixes the Cypher query returned by the `_get_hybrid_query` function, which is used by the hybrid retriever. Previously the scores returned with the nodes from vector search were not normalised correctly. The new query now matches the one found [here](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/neo4j_vector.py#L86
).

## Type of Change
- [ ] New feature
- [X] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [X] Unit tests
- [X] E2E tests
- [X] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [X] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

**Comment by willtai** [comment]
So now here each score calculated from both the vector and full-text indexes are normalized with respect to their maximum, and the reranked?

**Comment by alexthomas93** [comment]
> So now here each score calculated from both the vector and full-text indexes are normalized with respect to their maximum, and the reranked?
> 
> Were you able to manually test these results?

Yes, the scores from vector search are all divided by their maximum, and the scores from full-text search are all divided by their maximum. I have tested it yes.

File: src/neo4j_genai/neo4j_queries.py [hunk 1]
```diff
@@ -53,7 +53,9 @@
 def _get_hybrid_query() -> str:
     return (
         f"CALL {{ {VECTOR_INDEX_QUERY} "
-        f"RETURN node, score "
+        f"WITH collect({{node:node, score:score}}) AS nodes, max(score) AS max "
```
Comment (willtai) [inline src/neo4j_genai/neo4j_queries.py:56]: perhaps rename to `vector_index_max_score` and `ft_index_max_score` to be more explicit? I think it makes the Cypher query more readable

**Review by willtai — APPROVED** [review]
Approved with minor comments 🌳

**Comment by willtai** [comment]
@CodiumAI-Agent /update_changelog

**Pull request merged**